llvm/lib/Target/WebAssembly/WebAssemblyInstrSIMD.td

```diff
@@ -1199,6 +1213,7 @@ def : Pat<(v2f64 (froundeven (v2f64 V128:$src))), (NEAREST_F64x2 V128:$src)>;
 multiclass SIMDBinaryFP<SDPatternOperator node, string name, bits<32> baseInst> {
   defm "" : SIMDBinary<F32x4, node, name, baseInst>;
   defm "" : SIMDBinary<F64x2, node, name, !add(baseInst, 12)>;
+  defm "" : SIMDBinary<F16x8, node, name, !add(baseInst, 80), [HasHalfPrecision]>;
```

Comment:
I understand why it's added, and I wish we can multi-inherit from `SIMDBinary` and `HALF_PRECISION_I`, but I'm not sure if we can do it... (Can we?)

I'm not strongly opinionated about it and it's basically just a matter of preference, but how about adding a `multiclass` like `HalfPrecisionBinary` or something that inherits from `SIMDBinary`?

Reply:
I ended up adding `HalfPrecisionBinary`. I was hoping there was some way I could pass a multiclass id as a parameter so i could then pass in `SIMD_I` or `HALF_PRECISION_I` as an argument, but I couldn't figure out a way to make that work.